src/OAuth2.php

```diff
@@ -226,7 +226,8 @@ protected function generateAuthState()
      */
     protected function createToken(array $tokenConfig = [])
     {
-        $tokenConfig['tokenParamKey'] = 'access_token';
+        $defaultTokenConfig = ['tokenParamKey' => 'access_token'];
```

Comment:
Hmm, wait, should it be hardcoded? I was expecting to take same config set at the configuration of an app and to merge it with the incoming $tokenConfig.

Reply:
That's what it is doing, isn't it?

Reply:
The only thing that is merged by default is the class [here](https://github.com/yiisoft/yii2-authclient/blob/master/src/BaseOAuth.php#L217), there is no way to prepare the default config and the token itself is by default built on [the response](https://github.com/yiisoft/yii2-authclient/blob/master/src/OAuth2.php#L151). Token param key is set [here](https://github.com/yiisoft/yii2-authclient/blob/master/src/OAuthToken.php#L33) and it should be used to build the object, not the hardcoded value. Please correct if I'm wrong.

Reply:
The [OAuth2 Spec](https://www.rfc-editor.org/rfc/rfc6749#page-35) defines `access_token` as the parameter key. If i remember right `oauth_token` is from OAuth1. 

> Hmm, wait, should it be hardcoded? I was expecting to take same config set at the configuration of an app and to merge it with the incoming $tokenConfig.

Do you mean the config from the `authClientCollection`? I think you are right that i missed a point here, but it could be rather bad the way @KevinVanH propsed. We can get the config from the old `accessToken` but `getAccessToken()` potentially tries to refresh the old token. Since in this package the `createToken()` method is almost always used with the data from the Access-Token respone from the authorization server, this would be bad. `createToken()` is also called in the setter of `accessToken`, also a Possibility for loops.

I think a solution could be to add a `accessTokenConfig` parameter to `OAuth2` and use this parameter.

Reply:
Hmm... yes, seems you're right.